libs/algo/odc/algo/_percentile.py

```diff
@@ -46,9 +46,11 @@ def xr_percentile(
         
         for percentile in percentiles:
             name = f"{band}_pc_{int(100 * percentile)}"
-            data_vars[name] = da.map_blocks(
-                lambda x: np_percentile(x, percentile, nodata), xx.data, drop_axis=3
+            yy = da.map_blocks(
+                lambda x: np_percentile(x, percentile, nodata), xx, drop_axis=0, meta=np.array([], dtype=xx.dtype)
             )
+            
+            data_vars[name] = (("y", "x"), yy)
```

Comment:
unfortunately it's not always `y,x`. you are better off using `.dims` of input to keep names as they were.